### ISSUE TITLE
4.0.0-alpha-git.8b5f44d getting "no-auth-mech" when trying to connect

### ISSUE THREAD

**Issue by hnategh** [jsxc/jsxc#741]
### Expected behavior
I am tying to login via latest 4.0.0-alpha-git.8b5f44d and I get "no-auth-mech" error in client side.
### Steps to reproduce the behavior
1. Use index.html from example/
2. try to login via instance login form
3.

### Environment
- **JSXC version:**   4.0.0-git.00b15be
- **Host system and version:** jsxc example page Or using jsxc example-nightly/ 
- **Browser vendor and version:**  Chrome Version 70.0.3538.77
- **Any browser plugins enabled?** No
- **XMPP server vendor and version:** ejabberd 18.04 (tested also with provided server on example-nightly)
- **Is your XMPP server working with other clients as expected?** Yes, it works with other clients including the JSXC V3.0

#### Javascript

```
t {message: "no-auth-mech", errorCode: undefined}
```

#### JSXC

```
jsxc.bundle.js:11 [Warn] Instance of BaseErrors no-auth-mech
e.log @ jsxc.bundle.js:11
e.warn @ jsxc.bundle.js:11
(anonymous) @ jsxc.bundle.js:47
Promise.catch (async)
d @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
o @ jsxc.bundle.js:47
Promise.then (async)
s @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
i @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
t.start @ jsxc.bundle.js:47
(anonymous) @ example.js:33
dispatch @ jquery.min.js:3
q.handle @ jquery.min.js:3
example.js:37 >>> catch t {message: "no-auth-mech", errorCode: undefined}
jsxc.bundle.js:30 XHR finished loading: POST "https://www.jsxc.org/http-bind/".
c @ jsxc.bundle.js:30
_processRequest @ jsxc.bundle.js:30
_throttledRequestHandler @ jsxc.bundle.js:30
_connect @ jsxc.bundle.js:30
connect @ jsxc.bundle.js:30
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
t.login @ jsxc.bundle.js:47
e.connect @ jsxc.bundle.js:47
connect @ jsxc.bundle.js:47
d @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
o @ jsxc.bundle.js:47
Promise.then (async)
s @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
i @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
t.start @ jsxc.bundle.js:47
(anonymous) @ example.js:33
dispatch @ jquery.min.js:3
q.handle @ jquery.min.js:3
jsxc.bundle.js:11 [Warn] This action is only available in debug mode.
e.log @ jsxc.bundle.js:11
e.warn @ jsxc.bundle.js:11
t.deleteAllData @ jsxc.bundle.js:25
(anonymous) @ VM290:1
jsxc.bundle.js:11 [Debug] Instanciate otr for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate OMEMO for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate receipt for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate notification for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate meCommand for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate Message Archive Management for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate Chat State for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate httpUpload for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate vCard-based Avatars for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] Instanciate Carbon Copy for account demo9@demo.jsxc.ch
jsxc.bundle.js:11 [Debug] State changed to ESTABLISHING
jsxc.bundle.js:11 [Debug] Try to establish a new connection.
jsxc.bundle.js:11 [Debug] Strophe Connection Status:  CONNECTING
jsxc.bundle.js:11 [Debug] > <body rid=​"1708855279" xmlns=​"http:​/​/​jabber.org/​protocol/​httpbind" to=​"demo.jsxc.ch" xml:lang=​"en" wait=​"60" hold=​"1" content=​"text/​xml;​ charset=utf-8" ver=​"1.6" xmpp:version=​"1.0" xmlns:xmpp=​"urn:​xmpp:​xbosh">​</body>​
jsxc.bundle.js:11 [Debug] < <body xmlns=​"http:​/​/​jabber.org/​protocol/​httpbind" xmlns:xmpp=​"urn:​xmpp:​xbosh" xmlns:stream=​"http:​/​/​etherx.jabber.org/​streams" xmpp:version=​"1.0" authid=​"2913454530407416833" sid=​"ad96b6cec5ce155ec494b94e45b85c2fe7080ee5" wait=​"60" ver=​"1.11" polling=​"2" inactivity=​"30" hold=​"1" xmpp:restartlogic=​"true" requests=​"2" secure=​"true" maxpause=​"120" from=​"demo.jsxc.ch">​</body>​
jsxc.bundle.js:30 XHR finished loading: POST "https://www.jsxc.org/http-bind/".
c @ jsxc.bundle.js:30
_processRequest @ jsxc.bundle.js:30
_throttledRequestHandler @ jsxc.bundle.js:30
_connect @ jsxc.bundle.js:30
connect @ jsxc.bundle.js:30
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
t.login @ jsxc.bundle.js:47
e.connect @ jsxc.bundle.js:47
connect @ jsxc.bundle.js:47
d @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
o @ jsxc.bundle.js:47
Promise.then (async)
s @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
i @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
t.start @ jsxc.bundle.js:47
(anonymous) @ example.js:33
dispatch @ jquery.min.js:3
q.handle @ jquery.min.js:3
jsxc.bundle.js:11 [Warn] Instance of BaseErrors no-auth-mech
e.log @ jsxc.bundle.js:11
e.warn @ jsxc.bundle.js:11
(anonymous) @ jsxc.bundle.js:47
Promise.catch (async)
d @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
o @ jsxc.bundle.js:47
Promise.then (async)
s @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
i @ jsxc.bundle.js:47
(anonymous) @ jsxc.bundle.js:47
t.start @ jsxc.bundle.js:47
(anonymous) @ example.js:33
dispatch @ jquery.min.js:3
q.handle @ jquery.min.js:3
example.js:37 >>> catch 
```

#### XMPP

```
```

#### Host

```
```

Thank You!
Hadi

**Comment by sualko** [comment]
Can you send me some test credentials to klaus@jsxc.org? That would be awesome, because otherwise it's hard to investigate.

**Comment by sualko** [comment]
Any update on this?

**Comment by hnategh** [comment]
somehow I missed your comments on this issue. Anyway, I sent you an email with credentials. Thanks!

**Comment by sualko** [comment]
The issue was fixed in Strophe 1.2.16, see https://github.com/strophe/strophejs/issues/299
